feat: add ability to get namespace labels in k8s attributes action

BenElferink opened the pull request:
This can be extended in the UI-Kit easy to allow other Kinds, just let me know :)

<img width="666" height="288" alt="Screenshot 2025-09-03 at 14 46 46" src="https://github.com/user-attachments/assets/1767f746-ac67-49fe-8579-65add1a780b2" />

<img width="441" height="616" alt="Screenshot 2025-09-03 at 14 47 44" src="https://github.com/user-attachments/assets/7d96aa72-f314-4196-add9-2576b7f58816" />

<img width="550" height="196" alt="Screenshot 2025-09-03 at 14 50 21" src="https://github.com/user-attachments/assets/accd374f-0381-43a3-9994-f98287284dcf" />

emergency-ticket id: EMR-210

BenElferink committed:
feat: add 'from' field to K8sLabelAttribute and K8sAnnotationAttribute

- Introduced an optional 'from' field to both K8sLabelAttribute and K8sAnnotationAttribute types, allowing specification of the source (e.g., "pod" or "namespace").
- Updated deepcopy methods to handle the new 'from' field correctly.
- Modified CRD definitions and documentation to reflect the addition of the 'from' field with default value set to "pod".
- Adjusted related apply configuration files to support the new field.

BenElferink committed:
chore: update @odigos/ui-kit to version 0.0.101

- Bumped the @odigos/ui-kit dependency to version 0.0.101 in package.json and yarn.lock to incorporate the latest updates and improvements.

BenElferink committed:
Merge branch 'main' into ben/plat-308-add-ability-to-get-namespace-labels-as-part-of-the

BenElferink committed:
feat: add 'from' field to labels and annotations in k8sattributes documentation

- Updated the documentation for K8sLabelAttribute and K8sAnnotationAttribute to include the 'from' field, specifying its default value as "pod". This change enhances clarity on the source of attributes in Kubernetes.

RonFed commented on common/consts/consts.go:
```diff
@@ -140,3 +140,8 @@ const (
+
+// K8s attributes related consts
+const (
+	K8sAttributesFromDefaultValue = "pod"
```
nit: we can put this in a less global place, somewhere in autoscaler

RonFed commented on api/actions/v1alpha1/k8sattributes_types.go:
```diff
@@ -30,6 +30,11 @@ type K8sLabelAttribute struct {
+	// e.g. "pod" or "namespace"
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=pod
+	From *string `json:"from,omitempty"`
```
what do you think about using an "enum" here?
similar to 
```
// +kubebuilder:validation:Enum=pod,namespace
type K8sAttributeSource string

const (
	PodAttributeSource       K8sAttributeSource = "pod"
	NamespaceAAttributeSource     K8sAttributeSource = "namespace"
)
```

This way we'll have validation in the kube API level

RonFed reviewed (approved):
🐐 
left 2 small nits

BenElferink committed:
refactor: move K8sAttributesFromDefaultValue constant to k8sconsts package

- Removed the K8sAttributesFromDefaultValue constant from common/consts/consts.go and added it to api/k8sconsts/autoscaler.go.
- Updated references in k8sattributesresolver_controller.go to use the new location for the constant.

BenElferink committed:
feat: introduce K8sAttributeSource type and update related fields

- Added a new K8sAttributeSource type with enum values "pod" and "namespace" to represent the source of Kubernetes attributes.
- Updated K8sLabelAttribute and K8sAnnotationAttribute to use K8sAttributeSource for the 'from' field.
- Modified CRD definitions and documentation to reflect the new type and its possible values.
- Adjusted apply configuration and deepcopy methods to accommodate the changes.

BenElferink committed:
fix: correct namespace attribute source constant name

- Updated the constant name for the namespace attribute source from `NamespaceAAttributeSource` to `NamespaceAttributeSource` for consistency and clarity in the K8sAttributeSource type.

BenElferink committed:
Merge branch 'main' into ben/plat-308-add-ability-to-get-namespace-labels-as-part-of-the

BenElferink committed:
Merge branch 'main' into ben/plat-308-add-ability-to-get-namespace-labels-as-part-of-the

BenElferink committed:
Merge branch 'main' into ben/plat-308-add-ability-to-get-namespace-labels-as-part-of-the
